cicd/3-app/javabuilder/template.yml.erb

```diff
@@ -63,6 +63,7 @@ Globals:
     Tracing: Active
 Conditions:
   IsDevCondition: !Equals [!Ref BaseDomainName, "dev-code.org"]
+  IsDemoCondition: !Equals [!Ref SubdomainName, "javabuilder-demo"]
```

Comment:
I see you continued the existing pattern here, but we should try to keep this template environment-agnostic, and I would much rather refactor the design to allow us to eliminate the `IsDevCondition` rather than add a new `IsEnvironmentCondition`

You're using `IsDemoCondition` in two places, so I'll make comments directly there.

Reply:
PR to eliminate `IsDevCondition`, definitely a non-school-year project ;)
https://github.com/code-dot-org/javabuilder/pull/394